### ISSUE TITLE
[QA testing ticket] Raw /schedule_a/efile contributor name search not working

### ISSUE THREAD

**Issue by patphongs** [fecgov/openFEC#4085]
### Summary
@JonellaCulmer @lbeaufort @fecjjeng @PaulClark2 This original bug report can serve as the "testing" issue for the `TSVECTOR` work associated with `real_efile.sa7`. 

### Original reported issue
Below, you can see that "miller" is one of the contributor names:For raw schedule_a efile, the contributor_name field returns no results when searching by contributor_name.

**Completion criteria:**
- [ ] Confirm which fields are affected: seems to be contributor name only, so far
- [ ] testing idea: 
1. backfill real_efile.sa7_temporary_test_table with all the sa7 data without the trigger. Generate a "measure" of performance
2. backfill real_efile.sa7 (the actual table) with the trigger in place. Generate a measure of performance
3. Compare measure from 1 and 2

![Screen Shot 2019-11-12 at 11 57 42 AM](https://user-images.githubusercontent.com/12799132/68692595-c25c5480-0543-11ea-8d01-6a823c0534c6.png)

When "miller" is entered as a search criteria, there are no results.
Front end: https://www.fec.gov/data/receipts/?data_type=efiling&contributor_name=miller
API call: https://api.open.fec.gov/v1/schedules/schedule_a/efile/?api_key=DEMO_KEY&sort_hide_null=false&sort_nulls_last=false&data_type=efiling&contributor_name=miller&sort=-contribution_receipt_date&per_page=30&page=1

![Screen Shot 2019-11-12 at 11 58 16 AM](https://user-images.githubusercontent.com/12799132/68692688-e61f9a80-0543-11ea-8094-fd0d4ec7f2b4.png)

**Comment by jason-upchurch** [comment]
API example above:
https://api.open.fec.gov/v1/schedules/schedule_a/efile/?api_key=DEMO_KEY&sort_hide_null=false&sort_nulls_last=false&data_type=efiling&contributor_name=miller&sort=-contribution_receipt_date&per_page=30&page=1

api.open.fec:https://api.open.fec.gov/v1/schedules/schedule_a/efile/?sort_nulls_last=false&sort=-contribution_receipt_date&page=1&sort_null_only=false&per_page=20&contributor_name=miller&api_key=DEMO_KEY&sort_hide_null=false

**Comment by jason-upchurch** [comment]
query:

```sql
SELECT *
FROM real_efile.sa7 
LEFT OUTER JOIN ofec_committee_history_mv 
AS ofec_committee_history_mv_1 
ON real_efile.sa7.comid = ofec_committee_history_mv_1.committee_id 
AND EXTRACT(year FROM real_efile.sa7.create_dt) 
    + CAST(EXTRACT(year FROM real_efile.sa7.create_dt) AS INTEGER) %% 2 =
    ofec_committee_history_mv_1.cycle 
LEFT OUTER JOIN real_efile.reps AS reps_1 
ON real_efile.sa7.repid = reps_1.repid 
LEFT OUTER JOIN efiling_amendment_chain_vw AS efiling_amendment_chain_vw_1 
ON efiling_amendment_chain_vw_1.repid = reps_1.repid 
WHERE real_efile.sa7.contributor_name_text @@ to_tsquery('miller:*')
```

**Comment by jason-upchurch** [comment]
```sql
select distinct contributor_name_text from real_efile.sa7;
```
returns 1 result (empty string). It appears there are no non-empty instances for `contributor_name_text` in `real_efile.sa7`.

**Comment by jason-upchurch** [comment]
meanwhile, 

```sql
SELECT count(*) FROM real_efile.sa7 
LEFT OUTER JOIN ofec_committee_history_mv AS ofec_committee_history_mv_1 
ON real_efile.sa7.comid = ofec_committee_history_mv_1.committee_id 
AND EXTRACT(year FROM real_efile.sa7.create_dt) + CAST(EXTRACT(year FROM real_efile.sa7.create_dt) AS INTEGER) % 2 =ofec_committee_history_mv_1.cycle 
LEFT OUTER JOIN real_efile.reps AS reps_1 ON real_efile.sa7.repid = reps_1.repid 
LEFT OUTER JOIN efiling_amendment_chain_vw AS efiling_amendment_chain_vw_1 
ON efiling_amendment_chain_vw_1.repid = reps_1.repid;
```
returns 31,961,931 results.

**Comment by jason-upchurch** [comment]
@fecjjeng @PaulClark2 I found this issue has to do with no `TSVECTOR` entries in the table `real_efile.sa7`. I searched our repo and see we still have issue #3834 open from June 2019. I just assigned that issue to me and will put this issue as blocked by that one. If #3834 resolves I will close this issue. Let me know if any objections!

**Comment by jason-upchurch** [comment]
The particular issue in the bug report appears to be resolved, however, we still need to backfill data.

**Comment by jason-upchurch** [comment]
Moving to 11.2 for the time being as #4093 was moved to 11.2

**Comment by fecjjeng** [comment]
This ticket was the original ticket that reported this problem.  It is discovered that was due to the tsvector columns (contributor_name_text, contributor_employer_text, and contributor_occupation_text) in real_efile.sa7 are not populated.  In the process data (disclosure.fec_fitem_sched_a), the corresponding columns are populated by triggers during insert.  Issue #3834 (PR 4095) had added triggers to real_efile.sa7 so new data coming in will have these columns populated.  Historical data are backfilled by issue #4093.  
It had been verified after completion of issue #4093, all records had been backfilled in all three database (DEV, STG, and PRD).

The original front end link
https://www.fec.gov/data/receipts/?data_type=efiling&contributor_name=miller
and API call
https://api.open.fec.gov/v1/schedules/schedule_a/efile/?api_key=DEMO_KEY&sort_hide_null=false&sort_nulls_last=false&data_type=efiling&contributor_name=miller&sort=-contribution_receipt_date&per_page=30&page=1
had records returned as expected.  

The API calls to query on contributor_employer and contributor_occupation also returns with records found.
https://api.open.fec.gov/v1/schedules/schedule_a/efile/?contributor_occupation=student&api_key=DEMO_KEY&sort_nulls_last=false&sort_null_only=false&per_page=20&page=1&sort=-contribution_receipt_date&sort_hide_null=false

https://api.open.fec.gov/v1/schedules/schedule_a/efile/?api_key=DEMO_KEY&contributor_employer=state&sort_nulls_last=false&sort_null_only=false&per_page=20&page=1&sort=-contribution_receipt_date&sort_hide_null=false

**Comment by fecjjeng** [comment]
This issue is fixed via issue #3834 and #4093.  
Closed the ticket.
